src/stream_manager.rs

```diff
@@ -229,6 +231,77 @@ impl StreamManager {
         }
     }
 
+    ///
+    /// list scope
+    ///
+    #[pyo3(text_signature = "($self)")]
+    pub fn list_scope<'p>(&self, _py: Python<'p>) -> PyResult<Vec<String>> {
+        let controller = self.cf.controller_client();
+        let scope_result = list_scopes(controller);
+        futures::pin_mut!(scope_result);
+
+        let mut scope_vector = Vec::new();
+        // Used tokio::runtime::Runtime to block on the async code
+        let _result = tokio::runtime::Runtime::new().unwrap().block_on(async {
+            while let Some(sc) = scope_result.next().await {
+                scope_vector.push(sc);
+            }
+            Ok::<_, PyErr>(())
+        });
+        // Checks for errors in the async block
+        _result.map_err(|e| PyErr::new::<pyo3::exceptions::PyException, _>(IntoPy::<Py<pyo3::PyAny>>::into_py(e, _py)))?;
+        let mut scope_vector_act: Vec<String> = Vec::new();
+        for scope_val in scope_vector {
+            match scope_val {
+                Ok(scope) => {
+                    scope_vector_act.push(scope.name)
+                },
+                Err(e) => {
+                    println!("Exception encountered while fetching scope list : {}", e);
+                },
+            }
+        }
+        Ok(scope_vector_act)
+    }
+
+
+    ///
+    /// list streams
+    ///
+    #[pyo3(text_signature = "($self, scope_name)")]
+    pub fn list_stream<'p>(&self, scope_name: &str, _py: Python<'p>) -> PyResult<Vec<String>> {
+        let controller = self.cf.controller_client();
+        let stream_result = list_streams(
+            Scope {
+                name: scope_name.to_string(),
+            },
+            controller,
+        );
+
+        futures::pin_mut!(stream_result);
+        let mut stream_vector = Vec::new();
+        let _result = tokio::runtime::Runtime::new().unwrap().block_on(async {
+            while let Some(sc) = stream_result.next().await {
+                stream_vector.push(sc);
+            }
+            Ok::<_, PyErr>(())
+        });
+        // Checks for errors in the async block
+        _result.map_err(|e| PyErr::new::<pyo3::exceptions::PyException, _>(IntoPy::<Py<pyo3::PyAny>>::into_py(e, _py)))?;
+        let mut stream_vector_act: Vec<String> = Vec::new();
+        for stream_val in stream_vector {
+            match stream_val {
+                Ok(scoped_stream) => {
+                    stream_vector_act.push(scoped_stream.stream.name)
+                },
+                Err(e) => {
```

Comment:
Same here.
It may be useful to have a private method so as to not duplicate so much